src/main/java/mate/academy/springboot/swagger/controller/ProductController.java

```diff
@@ -0,0 +1,134 @@
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/products")
+@Tag(name = "product", description = "The Product Manager API")
+public class ProductController {
+    private final ProductService productService;
+    private final SortUtil sortUtil;
+    private final DtoMapper<ProductRequestDto, ProductResponseDto, Product> productMapper;
+
+    @Operation(summary = "Add Product", description = "Add Product to DB")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "201", description = "Product created",
+                    content = @Content(schema = @Schema(implementation = Product.class))),
+            @ApiResponse(responseCode = "400", description = "Invalid input"),
+            @ApiResponse(responseCode = "409", description = "Product already exists") })
+    @PostMapping
+    public ProductResponseDto add(@RequestBody ProductRequestDto productRequestDto) {
+        Product product = productService.save(productMapper.mapToModel(productRequestDto));
+        return productMapper.mapToDto(product);
+    }
+
+    @Operation(summary = "Get a Product by its id")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Found the Product",
+                    content = { @Content(mediaType = "application/json",
+                            schema = @Schema(implementation = Product.class)) }),
+            @ApiResponse(responseCode = "400", description = "Invalid id supplied",
+                    content = @Content),
+            @ApiResponse(responseCode = "404", description = "Product not found",
+                    content = @Content) })
+    @GetMapping("/{id}")
+    public ProductResponseDto get(@PathVariable Long id) {
+        return productMapper.mapToDto(productService.getById(id));
+    }
+
+    @Operation(summary = "Delete Product by its id")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Delete the Product",
+                    content = { @Content(mediaType = "application/json",
+                            schema = @Schema(implementation = Product.class)) }),
+            @ApiResponse(responseCode = "400", description = "Invalid id supplied",
+                    content = @Content),
+            @ApiResponse(responseCode = "404", description = "Product not found",
+                    content = @Content) })
```

Comment:
what will happen when you will call `repository.deleteById(id)` and product with such `id` does not exist? Will it throw exception?

Reply:
refactored